app/src/main/java/com/example/newsapp/data/repository/UserRepository.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.newsapp.data.repository
+
+
+
+import com.example.newsapp.data.dao.UserDao
+import com.example.newsapp.data.entity.User
+import javax.inject.Inject
+
+class UserRepository @Inject constructor(private val userDao: UserDao){
+    fun saveUser(user: User) {
```

Comment:
Since accessing the DB is potentially a long-running task that might freeze the main thread it makes sense to make the functions (saveUser, getUserInfo) suspend to be able to run them in a coroutine.